content/en/docs/concepts/storage/projected-volumes.md

```diff
@@ -71,18 +71,29 @@ A container using a projected volume source as a [`subPath`](/docs/concepts/stor
 volume mount will not receive updates for those volume sources.
 {{< /note >}}
 
-## SecurityContext interactions
+### serviceAccountToken and securityContext
 
 The [proposal](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/2451-service-account-token-volumes#proposal) for file permission handling in projected service account volume enhancement introduced the projected files having the the correct owner permissions set.
 
-### Linux
+#### Linux
 
-In Linux pods that have a projected volume and `RunAsUser` set in the Pod
-[`SecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context),
-the projected files have the correct ownership set including container user
-ownership.
+In some cases, Kubernetes applies a security optimization for the contents of `serviceAccountToken`
+volumes.
+When all containers in a pod have the same `runAsUser` set in their
+[`PodSecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context)
+or container
+[`SecurityContext`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-1),
+then the kubelet  ensures that the contents of the `serviceAccountToken` volume are owned by that user,
+and that every file has its permission mode set to `0600`.
 
-### Windows
+{{< note >}}
+If you add any {{< glossary_tooltip text="ephemeral containers" term_id="ephemeral-container" >}} to
+a Pod, those won't have been present when the pod started running on that node.
+Adding an ephemeral container to a pod does **not** change any volume permissions
+that were set when the pod was created.
```

Comment:
The technical content lgtm, this might be able to be rephrased more clearly. I'd suggest something like the following, but will defer to docs reviewers:

```
{{< glossary_tooltip text="Ephemeral containers" term_id="ephemeral-container" >}}
added to a Pod after it is created do *not* change volume permissions that were set when the pod was created.
If a Pod's `serviceAccountToken` volume permissions were set to `0600` because all other containers in the
Pod have the same `runAsUser`, ephemeral containers must use the same `runAsUser` to be able to read the token.
```

Reply:
```suggestion
{{< glossary_tooltip text="Ephemeral containers" term_id="ephemeral-container" >}}
added to a Pod after it is created do *not* change volume permissions that were
set when the pod was created.

If a Pod's `serviceAccountToken` volume permissions were set to `0600` because
all other containers in the Pod have the same `runAsUser`, ephemeral
containers must use the same `runAsUser` to be able to read the token.
```
(credit to @liggitt there)